cccatalog-api/cccatalog/api/views/site_views.py

```diff
@@ -36,8 +41,14 @@ class HealthCheck(APIView):
     swagger_schema = None
 
     def get(self, request, format=None):
-        return Response('', status=200)
-
+        es_conn = es
+        image_index_exists = es_conn.indices.exists(index=["image"])
```

Comment:
This is a good start! We also need to check the [health](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-health.html) of the index, however, not just whether it exists. The healthcheck tells us important information about the status of the cluster, like whether replica shards have been initialized, which can make or break the performance of the cluster in production.

Since the cluster will always have a status of `yellow` in local environments, there should be an environment variable containing a list of acceptable statuses for this healthcheck (for local testing, we could set it to `HEALTHCHECK_STATUSES=yellow,green` in `docker-compose.yml`, but leave the default to `green`, which will be used in production).

I think you mentioned problems with the healthcheck hanging; perhaps you set the `wait_for_status` parameter, which we don't want to use in the context of the server healthcheck (we only want to check the health at the current moment and react accordingly).

We're also going to need to figure out how to make the unit tests pass under these conditions. The problem here is that the [unit tests wait for the server to be healthy](https://github.com/creativecommons/cccatalog-api/blob/a18f75fccdd7345beff820dff4ee69604cd53748/.travis.yml#L26) before loading the test data, which will never happen now since we treat missing data as an unhealthy state. We need a way to distinguish between "started, but there's no data" and "started, but healthcheck=red". I suggest modifying the result returned by the healthcheck to reflect this, and then picking up that field in `.travis.yml`.

If the index is unhealthy:
```
GET /healthcheck
HTTP 500 Internal Server Error
Content-Type: application/json
{
    "error": "IndexUnhealthy",
    "detail": "The status of the image index is unhealthy.",
}
```

If the index doesn't exist:
```
GET /healthcheck
HTTP 500 Internal Server Error
Content-Type: application/json
{
    "error": "IndexMissing",
    "detail": "The image index does not exist.",
}
```

You could then modify the healthcheck command in `.travis.yml` to treat `IndexMissing` as an acceptable state. [jq](https://stedolan.github.io/jq/) is probably the best tool for this.

Reply:
Wow! 
Thanks for the further breakdown and explanations.
 Let me go back and work something out with the suggestions. 
Fingers crossed :)